.github/workflows/publish-module-api-doc.yaml

```diff
@@ -22,6 +29,7 @@ jobs:
             doxygen Doxyfile.API
 
       - name: Publish generated API documentation to GitHub Pages
-        uses: JamesIves/github-pages-deploy-action@4.4.0
+        uses: JamesIves/github-pages-deploy-action@v4.4.0
         with:
           folder: doc/
+        if: ${{ github.event_name == 'push' }}
```

Comment:
Nit: if you add the `workflow_dispatch` trigger, it should be `${{ github.event_name != 'pull_request' }}`.